src/Native/configure.cmake

```diff
@@ -175,13 +181,27 @@ check_function_exists(
     kqueue
     HAVE_KQUEUE)
 
-check_function_exists(
-    gethostbyname_r
-    HAVE_GETHOSTBYNAME_R)
+check_prototype_definition(
+     gethostbyaddr_r
+     "int gethostbyaddr_r(const void* addr, const socklen_t len, int type, struct hostent* result, char* buffer, size_t buflen, stuct hostent** entry, int* error)"
+     0
+     "sys/types.h;netdb.h"
```

Comment:
Have you confirmed that with this change Linux builds still set HAVE_GETHOSTBYADDR_R to 1?  I see some differences here that make me a bit nervous.  We can see in the CI logs when the CI system comes back up, too, but it's definitely something to look for.   A danger with polyfills like this is you can find yourself accidentally and silently falling back to the manually-implemented solution even when there's a better one built-in.